TESTS/netsocket/tcp/main.cpp

```diff
@@ -72,7 +73,17 @@ static void _ifup()
     NetworkInterface *net = NetworkInterface::get_default_instance();
     nsapi_error_t err = net->connect();
     TEST_ASSERT_EQUAL(NSAPI_ERROR_OK, err);
-    printf("MBED: TCPClient IP address is '%s'\n", net->get_ip_address() ? net->get_ip_address() : "null");
+    const char *address = net->get_ip_address();
+
+    printf("MBED: TCPClient IP address is '%s'\n", address ? address : "null");
+#define MESH 3
+#if MBED_CONF_TARGET_NETWORK_DEFAULT_INTERFACE_TYPE == MESH
+    uint8_t addr_hex[16];
+    stoip6(address, strlen(address), addr_hex);
+    if ((addr_hex[0] & 0xe0) != 0x20) {
```

Comment:
what are these why is there a check for address for mesh

To me this looks like bug in code that is circumvented in tests?

In applications we should wait for Global UP event and then everything should work

Reply:
@mikter , this PR is now only trying to improve error reporting due to mesh network in RAAS being heavily loaded and often unable to perform UPD tests. We can tell that this situation occurs by observing the IP address our DUT got. If we did not receive a global address, the test will fail anyway, so we just want to minimize the amount of unnecessary debugging and print a reasonable error message suggesting 

Now that you mentioned it, perhaps it is neater to wait for `GLOBAL_UP` event instead of checking if the address is global or not, but the essence of the changes remains the same - don't run the tests if the global connection failed and print a reasonable error message, instead.

@tymoteuszblochmobica , can we try observing the events instead of observing the IP address? I imagine we could set a blocking semaphore after `connect()` returns and only release it in a `GLOBAL_UP` event callback. If the semaphore is not released in time, the test will timeout. If the last message printed is "Waiting for GLOBAL_UP" I think that will be a clear indication that there is a connectivity problem. Does this make sense?

Reply:
Replaced with GLOBAL_UP wait.